### PR TITLE
no CotW, but at least some notable changes

### PR DESCRIPTION
**Pull request by llogiq**

### CODE REVIEW

File: drafts/2016-12-13-this-week-in-rust.md [hunk 1]
```diff
@@ -47,39 +47,25 @@ If you are a Rust project owner and are looking for contributors, please submit
...
-* [Cargo will now correctly retry downloading in case of network error](https://github.com/rust-lang/cargo/pull/3348)
-* [Cargo has release branches now](https://github.com/rust-lang/cargo/pull/3345)
-* [Cargo: Fixed SSL paths](https://github.com/rust-lang/cargo/pull/3342) (this broke downloading for some days in nightly)
+74 pull requests were [merged in the last week][merged]. This contains a good number of plugin-breaking changes.
+
+[merged]: https://github.com/issues?q=is%3Apr+org%3Arust-lang+is%3Amerged+merged%3A2016-12-05..2016-12-12
+
+* [Use link(kind) annotation to fix native Windows imports](https://github.com/rust-lang/rust/pull/37973) ([RFC #1717](https://github.com/rust-lang/rfcs/blob/master/text/1717-dllimport.md))
+* [Function arity errors now show the original definition](https://github.com/rust-lang/rust/pull/38121)
+* [`HashMap`/-`Set` now allocate smarter on `extend_from_iter(_)`](https://github.com/rust-lang/rust/pull/38017)
```
Comment (bluss) [inline drafts/2016-12-13-this-week-in-rust.md]: `extend` and `from_iter`

File: drafts/2016-12-13-this-week-in-rust.md [hunk 2]
```diff
@@ -47,39 +47,25 @@ If you are a Rust project owner and are looking for contributors, please submit
...
+* [More forwarded `ExactSizeIterator` / `is_empty()`](https://github.com/rust-lang/rust/pull/38149)
+* [Stricter lifetimes for `LateLintPass`](https://github.com/rust-lang/rust/pull/38191) (Plugin-unbreaking 🙂)
+* [`Iterator::nth(_)` no longer needs `self` to be `Sized`](https://github.com/rust-lang/rust/pull/38134)
+* [Incremental compilation will now recompile items on visibility change](https://github.com/rust-lang/rust/pull/38272)
+* [`-Z always_encode_mir`](https://github.com/rust-lang/rust/pull/38217)
+* [dylib symbol handling improvements](https://github.com/rust-lang/rust/pull/38117)
+* [`tidy` now checks the licenses of vendored dependencies](https://github.com/rust-lang/rust/pull/38291)
+* [Rustbuild is now the default build system](https://github.com/rust-lang/rust/pull/37817) (1 year in the making)
+* [Allow `--test` on procedural-macro crates](https://github.com/rust-lang/rust/pull/38107)
+* [Cargo now picks up `build.rs` scripts by default](https://github.com/rust-lang/cargo/pull/3361) (unless `package.build = false` explicitly)
```
Comment (bluss) [inline drafts/2016-12-13-this-week-in-rust.md]: Cool but they didn't pull the trigger on it. Cargo will pick them up in the future, for now an orphan build.rs is a warning (and no build script).

**Comment by bluss** [comment]
I think the sort improvement is really exciting. Just don't know how to mark it.

**Comment by llogiq** [comment]
Agreed. Perhaps something like "epic perf improvement"?

File: drafts/2016-12-13-this-week-in-rust.md [hunk 3]
```diff
@@ -47,39 +47,25 @@ If you are a Rust project owner and are looking for contributors, please submit
...
-* [Cargo will now correctly retry downloading in case of network error](https://github.com/rust-lang/cargo/pull/3348)
-* [Cargo has release branches now](https://github.com/rust-lang/cargo/pull/3345)
-* [Cargo: Fixed SSL paths](https://github.com/rust-lang/cargo/pull/3342) (this broke downloading for some days in nightly)
+74 pull requests were [merged in the last week][merged]. This contains a good number of plugin-breaking changes.
+
+[merged]: https://github.com/issues?q=is%3Apr+org%3Arust-lang+is%3Amerged+merged%3A2016-12-05..2016-12-12
+
+* [Use link(kind) annotation to fix native Windows imports](https://github.com/rust-lang/rust/pull/37973) ([RFC #1717](https://github.com/rust-lang/rfcs/blob/master/text/1717-dllimport.md))
+* [Function arity errors now show the original definition](https://github.com/rust-lang/rust/pull/38121)
+* [`HashMap`/-`Set` now allocate smarter on `extend(_)`, `from_iter(_)`](https://github.com/rust-lang/rust/pull/38017)
```
Comment (bluss) [inline drafts/2016-12-13-this-week-in-rust.md]: Sorry for not catching it in the first round by the way. I think `from_iter` ended up unchanged, only extend switched "policy"

**Comment by bluss** [comment]
:100: Epic

**Pull request merged**